api/src/main/java/org/openmrs/module/pihcore/task/PihRemovePatientsFromMCOEQueue.java

```diff
@@ -88,11 +88,15 @@ private int removeTimedOutPatients(Queue queue) {
         for (QueueEntry entry : queueEntries) {
             Date startedAt = entry.getStartedAt();
             long activeMinutes = Duration.between(startedAt.toInstant(), currentTime.toInstant()).toMinutes();
-            log.warn("Removing patient {} from queue after {} minutes", entry.getPatient().getUuid(), activeMinutes);
             entry.setEndedAt(currentTime);
-            queueServices.getQueueEntryService().saveQueueEntry(entry);
+            try {
+                queueServices.getQueueEntryService().saveQueueEntry(entry);
+            } catch (Exception e) {
+                log.error("Failed to remove patient: " + entry.getPatient().getUuid() + " from queue.", e);
+                continue;
+            }
+            log.warn("Removing patient {} from queue after {} minutes", entry.getPatient().getUuid(), activeMinutes);
             numPatientsRemoved++;
```

Comment:
Personally I don't like using the `continue;` construct, as it is confusing and hard to nest.  Can we instead just move these 2 statements into the try block, and then if an exception is caught we simply log that?